frontend/src/views/pages/transactions/TransactionDetails.vue

```diff
@@ -0,0 +1,325 @@
+      axios({
+          method: 'PUT',
+          url: `${this.$store.state.api}/inventory/details/sale/${this.saleDetail.id}`,
+          headers: {
+            'Authorization': `Bearer ${this.$store.state.auth.token}`
+          },
+          data: {
+            date_approved: date,
+          }
+      }).then(response => {
+          if (response.data.success) {
+            this.saleDetail['Approval Date'] = date;
+            this.setNewSale(this.saleDetail);
+            this.setTransactionModal(false);
+            this.showMessage("The sale has been approved", "success");
+          }
+      }).catch(error => {
+          console.log(error);
+          this.showMessage(error.response.data.error, "danger");
+      })
+    },
+    getSale() {
+      console.log(this.saleDetail);
+        axios({
+        method: "GET",
+        url: `${this.$store.state.api}/inventory/details/sale/${this.saleDetail.id}`,
+        headers: {
+          Authorization: `Bearer ${this.$store.state.auth.token}`,
+        },
+      })
+        .then((response) => {
+          if (response.data.success) {
+            this.saleObject = response.data.payload;
+            this.form = this.getEmptyForm();
+          }
+        })
+        .catch((err) => {
+          console.log(err);
+        });
+    },
+    markVehicleDelivered() {
+      let body = this.saleObject.vehicle;
+      let ts = Date.now();
+      let date_ob = new Date(ts);
+      let date = date_ob.getFullYear() + "-" + date_ob.getMonth() + 1 + "-" + date_ob.getDate();
+      body.delivered = true;
+      body.date_delivered = date;
+      axios({
+        method: "PUT",
+        url: `${this.$store.state.api}/inventory/vehicle/${body._id}`,
+        headers: {
+          Authorization: `Bearer ${this.$store.state.auth.token}`,
+        },
+        data: body,
+      })
+        .then((response) => {
+          if (response.data.success) {
+            console.log(response.data.payload);
+            this.saleDetail['Delivery Status'] = "Delivered";
+            this.saleDetail['Delivery Date'] =  date;
+            this.setNewSale(this.saleDetail);
+            this.setTransactionModal(false);
+            this.showMessage("Vehicle has been marked as delivered", "success");
+          }
+        })
+        .catch((err) => {
+          console.log(err);
+          this.showMessage("Error occured while updating vehicle delivery status", "danger");
+        });
+    },
+    updateSale() {
+      let body = this.form;
+      axios({
+          method: 'PUT',
+          url: `${this.$store.state.api}/inventory/details/sale/${this.saleDetail.id}`,
+          headers: {
+              'Authorization': `Bearer ${this.$store.state.auth.token}`
+          },
+          data: body
+      }).then(response => {
+          if (response.data.success) {
+            this.saleDetail['Deposit'] = response.data.payload.deposit_amount;
+            this.setNewSale(this.saleDetail);
```

Comment:
Instead of setting the sales table as a new object from the response, you can simply call the fetchSales() method that is passed down from the TransactionTable parent component. This way the table refreshes with the proper formatting.

Reply:
This is a good idea. The code has been updated to follow this suggestion in this [commit](https://github.com/purplesmurf1998/Autotracks/pull/52/commits/3e369c5152623fbe180c6911d67c2c09cb5b502f)